### DISCUSSION TITLE
tweak ui scale globally & how to set font size

### DISCUSSION THREAD

**Discussion by adsick** [emilk/egui#172] [Q&A] [answered]
Hi, I want to have access to scale of everything so I can make ui larger and easier to work around.
Tweaking font size is also ambiguous for me now, I just tried to "reverse eng" where values of font sizes lie in the source code by walking from ui.settings_ui() to strange stuff named FontDefinitions. 
so the second question is how to tweak font sizes in code?

**Comment by emilk** [comment, marked as answer]
If you want to scale the entire GUI use the `pixels_per_point` option (the native example app has a slider for gui scale). For the web version just zoom the web page (like you would any web page).

To change the fonts use [`set_fonts`](https://docs.rs/egui/0.9.0/egui/struct.Context.html#method.set_fonts). In the example for `FontDefinitions` you can see how the font size for buttons is set: https://docs.rs/egui/0.9.0/egui/struct.FontDefinitions.html

**Reply by adsick** [reply]
> 
> 
> If you want to scale the entire GUI use the `pixels_per_point` option (the native example app has a slider for gui scale).
> 
I see what are you talking about, but I don't understand how do I access that ```pixels_per_point``` thing (in a mutable way of course) all I see for now is my ```ctx: &CtxRef``` and it's ```.pixels_per_point()``` method which doesn't work out when I want to change the value.

it would be nice to have some kind of ```.set_scale(...)``` method on context.

**Reply by emilk** [reply]
I added `Context::set_pixels_per_point` in https://github.com/emilk/egui/commit/5f6a4688128d9f86a37405e87735b2b2d706e6dc

**Reply by buxx** [reply]
@emilk Hello, about "For the web version just zoom the web page (like you would any web page)" is it possible to programmatically change the egui pixels_per_point value in web context ? I work on a game playable on mobile device, I want to zoom only egui integration and at the init of the game.

**Reply by emilk** [reply]
I don't think so at the moment. You could perhaps add a setting to `eframe::WebOptions` to control if we should follow the web page zoom level or not

**Comment by frankvgompel** [comment]
Could anyone explain this to me? For instance where do I put Context::set_pixels_per_point() in the "Hello world" example:

https://github.com/emilk/egui/blob/master/examples/hello_world/src/main.rs

And what is the default value of this?

**Reply by buxx** [reply]
Just put it at line before ui.heading, like this :+1: 

```rust
impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::CentralPanel::default().show(ctx, |ui| {
            ctx.set_pixels_per_point(5.0);
            ui.heading("My egui Application");
            ui.horizontal(|ui| {
                ui.label("Your name: ");
                ui.text_edit_singleline(&mut self.name);
            });
            ui.add(egui::Slider::new(&mut self.age, 0..=120).text("age"));
            if ui.button("Click each year").clicked() {
                self.age += 1;
            }
            ui.label(format!("Hello '{}', age {}", self.name, self.age));
        });
    }
}
```

I guess default value is 1.0

**Reply by frankvgompel** [reply]
Thanks.

**Comment by steadmon** [comment]
Regarding `set_pixels_per_point`, is there a way to have it take effect before the first frame is rendered? I call `ctx.set_pixels_per_point(1.0)` as the very first statement in `update`, but I still get a single frame rendered with 1.6 PPP, which seems to be coming from `integration_info.native_pixels_per_point`. I was hoping I could override this by calling `set_pixels_per_point` in the `AppCreator`, but it doesn't have any effect.

**Reply by buxx** [reply]
Sorry, I don't know. I let author or someone else trying to answer this question.

**Reply by sstickl** [reply]
I know this is old, but for others that are trying to figure out the same problem:

If you call it in your app creation area, it will change the size before the first frame.

Example:

    eframe::run_native(
        "App",
        native_options,
        Box::new(|cc: &eframe::CreationContext<'_>| {
            cc.egui_ctx.set_pixels_per_point(1.2);
            Ok(Box::<app::App>::default())
        }),
    )

**Reply by aurexav** [reply]
> ```
> eframe::run_native(
>     "App",
>     native_options,
>     Box::new(|cc: &eframe::CreationContext<'_>| {
>         cc.egui_ctx.set_pixels_per_point(1.2);
>         Ok(Box::<app::App>::default())
>     }),
> )
> ```

This won't work. I need to set this property after the app is fully started.

**Reply by buxx** [reply]
With eframe 0.31.1, I do the following : 

```rust
impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        ctx.set_zoom_factor(2.0);
    // [...]
```

**Comment by RDMCz** [comment]
Hi, if someone ended up here trying to find a way to set text size for individual elements, you can use `RichText`:

```rust
ui.label(egui::RichText::new("Big label").size(140.0));
let _ = ui.button(egui::RichText::new("Big button").size(140.0));
```

This might be little bit off-topic, but it took me some time to find it, and Google kept suggesting me this discussion as the first result. So, maybe it will help someone in the future.
